hopp/simulation/technologies/sites/site_info.py

```diff
@@ -25,140 +37,121 @@ def plot_site(verts, plt_style, labels):
+    Represents site-specific information needed by the hybrid simulation class and layout optimization.
 
-    Attributes
-    ----------
-    data : dict 
-        dictionary of initialization data
-    lat : float
-        site latitude [decimal degrees]
-    long : float
-        site longitude [decimal degrees]
-    vertices : np.array
-        site boundary vertices [m]
-    polygon : shapely.geometry.polygon
-        site polygon
-    valid_region : shapely.geometry.polygon
-        `tidy` site polygon
-    solar_resource : :class:`hybrid.resource.SolarResource`
-        class containing solar resource data
-    wind_resource : :class:`hybrid.resource.WindResource`
-        class containing wind resource data
-    elec_prices : :class:`hybrid.resource.ElectricityPrices`
-        Class containing electricity prices
-    n_timesteps : int
-        Number of timesteps in resource data
-    n_periods_per_day : int
-        Number of time periods per day
-    interval : int
-        Number of minutes per time interval 
-    urdb_label : string
-        `Link Utility Rate DataBase <https://openei.org/wiki/Utility_Rate_Database>`_ label for REopt runs
-    capacity_hours : list
-        Boolean list where ``True`` if the hour counts for capacity payments, ``False`` otherwise
-    desired_schedule : list
-        Absolute desired load profile [MWe]
-    follow_desired_schedule : boolean
-        ``True`` if a desired schedule was provided, ``False`` otherwise
+    Args:
+        data (dict): Dictionary containing site-specific information.
+        solar_resource_file (Union[Path, str], optional): Path to solar resource file. Defaults to "".
+        wind_resource_file (Union[Path, str], optional): Path to wind resource file. Defaults to "".
+        grid_resource_file (Union[Path, str], optional): Path to grid pricing data file. Defaults to "".
+        hub_height (float, optional): Turbine hub height for resource download in meters. Defaults to 97.0.
+        capacity_hours (:obj:`NDArray`, optional): Boolean list indicating hours for capacity payments. Defaults to [].
+        desired_schedule (:obj:`NDArray`, optional): Absolute desired load profile in MWe. Defaults to [].
+        solar (bool, optional): Whether to set solar data for this site. Defaults to True.
+        wind (bool, optional): Whether to set wind data for this site. Defaults to True.
     """
+    # User provided
+    data: dict
+    solar_resource_file: Union[Path, str] = field(default="", converter=resource_file_converter)
+    wind_resource_file: Union[Path, str] = field(default="", converter=resource_file_converter)
+    grid_resource_file: Union[Path, str] = field(default="", converter=resource_file_converter)
+    hub_height: hopp_float_type = hopp_float_type(97.)
+    capacity_hours: NDArray = field(default=[], converter=converter(bool))
+    desired_schedule: NDArrayFloat = field(default=[], converter=converter())
+    solar: bool = True
+    wind: bool = True
```

Comment:
Should `solar` and `wind` have `field` as well?

Reply:
It's not required, I left it simple because I couldn't think of a reason we'd need a validator/converter on there, but I'm fine to add it back if you prefer

Reply:
I think it would be nice to include the field method, as these will eventually be raised to the input file level. Then can still be optional.